selfdrive/ui/qt/widgets/cameraview.cc

```diff
@@ -296,13 +296,16 @@ void CameraViewWidget::vipcConnected(VisionIpcClient *vipc_client) {
 #ifdef QCOM2
   egl_display = eglGetCurrentDisplay();
 
-  for (auto &pair : egl_images) {
-    eglDestroyImageKHR(egl_display, pair.second);
+  for (auto &[_, s] : egl_images) {
+    eglDestroyImageKHR(egl_display, s.handle);
+    close(s.fd);
   }
   egl_images.clear();
-
   for (int i = 0; i < vipc_client->num_buffers; i++) {  // import buffers into OpenGL
-    int fd = dup(vipc_client->buffers[i].fd);  // eglDestroyImageKHR will close, so duplicate
+    // the EGL_EXT_image_dma_buf_import spec was revised after 2013:
+    // EGL does not take ownership of the file descriptors. It is the
+    // responsibility of the application to close it.
+    int fd = dup(vipc_client->buffers[i].fd);
```

Comment:
Still using dup to make sure it works under the old spec. It should be harmless to close fd twice.